docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,305 @@
+
+10. `Add inferred types`
+
+All other nodes not already defined are inspected and have types [inferred](/docs/schema-customization/#automatic-type-inference) by Gatsby.
+
+11. `Processing types`
+
+Composes 3rd party schema types, children fields, custom resolve functions, set fields, and print type definitions into the GraphQL schema.
+
+12. `building schema`
+
+Imports the composed GraphQL schema and builds it.
+
+13. `createPages`
+
+Calls the [`createPages` API](/docs/node-apis/#createPages) for your site and all plugins implementing it, like when you [create pages programatically](/docs/programmatically-create-pages-from-data/) in your `gatsby-node.js`.
+
+Plugins can handle the [`onCreatePage` event](/docs/node-apis/#onCreatePage) at this point for use cases like manipulating the path of pages.
+
+14. `createPagesStatefully`
+
+Similar to the `createPages` step, but for the [`createPagesStatefully` API](/docs/node-apis/#createPagesStatefully).
+
+15. `onPreExtractQueries`
+
+Calls the [`onPreExtractQueries` API](/docs/node-apis/#onPreExtractQueries) for your site and all plugins implementing it.
+
+16. `update schema`
+
+Rebuilds the GraphQL schema, this time with `SitePage` context.
+
+17. `extract queries from components`
+
+All JavaScript files in the site are loaded and Gatsby determines if there are any GraphQL queries in them. If there are problematic queries they can be reported back with warnings or errors. All these queries get queued up for execution in a later step.
+
+18. `write out requires`
+
+An internal Gatsby utility adds the code that files need to load/require.
+
+19. `write out redirect data`
+
+An internal Gatsby utility that adds code for redirects, like implemented with [`createRedirect`](/docs/actions/#createRedirect).
+
+20. `Build manifest and related icons` - (from `gatsby-plugin-manifest`)
+
+This step is actually activated by `gatsby-plugin-manifest` in the `gatsby-default-starter` and is not a part of the built-in Gatsby functionality, demonstrating plugins are able to tap into the lifecycle.
+
+21. `onPostBootstrap`
+
+Calls the [`onPostBootstrap` API](/docs/node-apis/#onPostBootstrap) for your site and all plugins implementing it.
+
+### Steps of the build phase
+
+1. `run static queries`
+
+Static queries that were queued up earlier from query extraction are actually run so the data pages need can be provided to them.
+
+2. `Generating image thumbnails — 6/6` - (from `gatsby-plugin-sharp`)
+
+Another step that is not a part of the built-in Gatsby functionality, but is the result of installing `gatsby-plugin-sharp`, which taps into the lifecycle. Sharp runs processing on images to create thumbnails of different sizes.
+
+3. `Building production JavaScript and CSS bundles`
+
+Compiles JavaScript and CSS using webpack.
+
+4. `Rewriting compilation hashes`
+
+Compilation hashes are used by webpack, and all files with page data need to be updated with the new hashes since they've been recompiled.
```

Comment:
```suggestion
Compilation hashes are used by webpack to keep the cache up to date, and all files with page data need to be updated with the new hashes since they've been recompiled.
```
A short explanation of why this is used would help: e.g. is it for invalidation of the cache or security purposes (or both)?